app/main.py

```diff
@@ -1600,6 +1605,37 @@ def update_alert_frame_main(alert_frame_update_new_event, alert_frame_update_int
         return alert_frame_update_interval, dash.no_updates
 
 
+# ----------------------------------------------------------------------------------------------------------------------
+# Callbacks related to the "dashboard_screen" page
+
+@app.callback(
+    Output('dashboard_table', 'children'),
+    Input('interval-component-dashboard-screen', 'n_intervals'), )
+def update_dashboard_table(n_intervals):
+    """
+    This builds and refreshes the dashboard table to monitor devices status every minute
+    Had to fetch devices data again for freshness purposes
+    """
+    response = requests.get('https://api.pyronear.org/devices/', headers=api_client.headers)
+    # Check token expiration
+    if response.status_code == 401:
+        api_client.refresh_token(cfg.API_LOGIN, cfg.API_PWD)
+        response = requests.get('https://api.pyronear.org/devices/', headers=api_client.headers)
+
+    # We filters devices_data to only display devices belonging to the sdis and then make the comparison between
+    # last_ping and datetime.now()
+    all_devices = pd.DataFrame(response.json())
+
+    sdis_devices = all_devices.sort_values('login').loc[all_devices['id'].isin(list(range(2, 18)))].drop(
+        columns=['id', 'created_at', 'pitch', 'elevation', 'specs', 'angle_of_view', 'software_hash', 'owner_id'])
+    sdis_devices['last_ping_hours_dif'] = sdis_devices['last_ping'].apply(lambda x: (pd.to_datetime(x) - pd.to_datetime(
+        datetime.utcnow().isoformat())).total_seconds() // 3600)
```

Comment:
These two lines are slightly difficult to read. Could you very briefly add some indentations? Sorry for the very low value-added comment 😬

Reply:
I know this is kinda heavy, will try some indent